File: php/features.php
<?php
declare(strict_types=1);

// JokeDelivery SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class JokeDeliveryFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new JokeDeliveryBaseFeature();
            case "test":
                return new JokeDeliveryTestFeature();
            default:
                return new JokeDeliveryBaseFeature();
        }
    }
}

File: py/features.py
# JokeDelivery SDK feature factory

from feature.base_feature import JokeDeliveryBaseFeature
from feature.test_feature import JokeDeliveryTestFeature


def _make_feature(name):
    features = {
        "base": lambda: JokeDeliveryBaseFeature(),
        "test": lambda: JokeDeliveryTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
